src/services/KeywordExtractionService.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * Extracts keywords of given PDF document. Keywords are
+ * extracted from metadata of the PDF or, if they are not
+ * found there, they are extracted from 'Keywords' section
+ * in text of PDF document. If no keywords are found, empty
+ * array is returned.
+ * @async
+ * @param {PDFDocumentProxy} pdfDoc PDF document
+ * @returns {String[]} keywords array
+ */
+async function extractKeywords(pdfDoc) {
+  let keywords = await getKeywordsFromMetadata(pdfDoc);
+
+  if (!keywords) keywords = await getKeywordsFromText(pdfDoc);
+
+  console.log(keywords); //TODO: remove
+
+  return keywords;
+}
+
+/**
+ * Extracts keywords from metadata of given PDF document.
+ * If keywords are not in the PDF document undefined returned.
```

Comment:
It is probably better to return explicitely NULL, instead of return nothing, I know it is the same, but it is more readable.

Reply:
I actually had in mind returning an empty array since it has a better semantic meaning than null/undefined.